libsolidity/interface/FileReader.cpp

```diff
@@ -269,7 +269,8 @@ boost::filesystem::path FileReader::normalizeCLIPathForVFS(
 	if (!isUNCPath(normalizedPath))
 	{
 		boost::filesystem::path workingDirRootPath = canonicalWorkDir.root_path();
-		if (normalizedRootPath == workingDirRootPath)
+		// Ignore drive letter case on Windows (C:\ <=> c:\).
+		if (boost::filesystem::equivalent(normalizedRootPath, workingDirRootPath))
 			normalizedRootPath = "/";
```

Comment:
The thing is, we completely ignore case-insensitivity right now and drive letters are not an exception. You'll for example get into problems if you have different case in `--base-path` and the paths you use in file paths. And this is not specific to case - you can have problems using chars not allowed on some filesystems or stuff that depends on unicode equivalence.

Originally I wanted the normalization to include using the canonical capitalization (both NTFS on Windows and HFS on macOS are case-preserving and the case-insensitivity is a layer above that) but unfortunately Boost does not provide anything that would let me check what it is so I dropped the idea. I think that something like #11412 might be a better and more general way to approach the problem.

Case normalization specifically for root path is *probably* ok (the only two cases with named roots I'm aware of are drive letters and UNC paths) but I'd rather not add such specific hacks unless this is actually impacting some practical case. Can you run into it by running a `solc` command that has consistent capitalization? Or by just running tests? Or does it require doing something unusual?

Reply:
Perhaps I should explain how I came here.

On my Windows system I have `soltest` on `D:` and the temporary folder on `C:`. That leads to failures like:

```
error: in "CommandLineInterfaceTest/cli_ignore_missing_some_files_exist": check result.reader.sourceUnits() == expectedSources has failed [{
C:/Users/ . . . . /input1.sol: ,
} != {
/Users/ . . . . /input1.sol: ,
}]
```

I can workaround it by changing `TMP` like `set TMP=D:\something`, but not like `set TMP=d:\something` (small `d`).

Hence the proposal. But if it creates more troubles than fixes (and the path normalization code already looks fragile), we can just extend the "contributing" docs correspondingly.

What do you think?

Reply:
Thanks for the use case. That actually helps a lot. It's only a special case but looks like it could be common enough to be worth fixing.

> On my Windows system I have `soltest` on `D:` and the temporary folder on `C:`. That leads to failures

That seems to actually be a mistake in these tests. They assume that temp is always on the same drive and that's true in our CI but wrong in general. I need to fix them.